data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessor.java

```diff
@@ -48,20 +51,28 @@ public TranslateProcessor(PluginMetrics pluginMetrics, final TranslateProcessorC
         this.expressionEvaluator = expressionEvaluator;
         individualMappings = new HashMap<>();
         rangeMappings = new LinkedHashMap<>();
-        if(this.translateProcessorConfig.getRegexParameterConfiguration()!=null) {
-            patternMappings = translateProcessorConfig.getRegexParameterConfiguration().getPatterns();
-        }
-        else{
-            patternMappings = Collections.emptyMap();
+        patternMappings = new HashMap<>();
+        compiledPatterns = new HashMap<>();
+        if (Objects.nonNull(this.translateProcessorConfig.getRegexParameterConfiguration())) {
+            compilePatterns(translateProcessorConfig
+                                    .getRegexParameterConfiguration()
+                                    .getPatterns());
         }
-
         processMapField(translateProcessorConfig.getMap());
         parseFile(translateProcessorConfig.getFilePath());
         checkOverlappingKeys();
     }
 
-    private void processMapField(Map<String, String> map){
-        if(Objects.nonNull(map)) {
+    private void compilePatterns(Map<String, String> mappings) {
+        patternMappings.putAll(mappings);
```

Comment:
Is the patternMappings object still necessary? Isn't it a duplicate essentially of the compiledPatterns?

Reply:
I want to check for substring also when pattern fails. Original strings are in patternMappings.
pattern.matcher(sourceKey).find() checks for pattern inside of sourceKey but not other way around. Is there a way to do it? I will investigate more on this and will let you know. Must be possible